common/persistence/sql/execution_util.go

```diff
@@ -1165,7 +1165,7 @@ func updateExecution(
 func workflowExecutionStateFromCurrentExecutionsRow(
 	row *sqlplugin.CurrentExecutionsRow,
 ) (*persistencespb.WorkflowExecutionState, error) {
-	if len(row.Data) > 0 {
+	if len(row.Data) > 0 && row.DataEncoding != "" {
```

Comment:
maybe len(raw.DataEncoding) > 0 to handle nil case as well?

Reply:
`DataEncoding` is a string, so comparing with empty string is the same as comparing the size `> 0`.